### ISSUE TITLE
Issues with PyTorch models using BatchNormalization

### ISSUE THREAD

**Issue by jafermarq** [adap/flower#611]
In PyTorch models making use of BatchNormalization, there's in issue when executing `set_parameters()` in the client. This is because the global model was initialised using the parameters of a model (those from the first connected client) that hadn't seen any input batches yet. Because of this, when calling `get_parameters()` in the first client, the internal variable `num_batches_tracked` of batchnorm layers will have a `numpy` of shape `()`, causing the error later on when trying to initialise the rest of the clients with `set_params()`. This is because `num_batches_tracked` is initialised as `tensor(0)`, which causes the issue.

This issue can be replicated using the [quickstart_pytorch](https://github.com/adap/flower/tree/main/examples/quickstart_pytorch) example and adding a `nn.BatchNorm2d` after one of the `nn.Conv2d` layers. If adding this to the second one, the shapes of `net.state_dict()` are:

```
conv1.weight torch.Size([6, 3, 5, 5])
conv1.bias torch.Size([6])
conv2.weight torch.Size([16, 6, 5, 5])
conv2.bias torch.Size([16])
bn.weight torch.Size([16])
bn.bias torch.Size([16])
bn.running_mean torch.Size([16])
bn.running_var torch.Size([16])
bn.num_batches_tracked torch.Size([]) <--- issue. Set_params() will try to set shape torch.Size([0]) instead of torch.Size([])
fc1.weight torch.Size([120, 400])
fc1.bias torch.Size([120])
fc2.weight torch.Size([84, 120])
fc2.bias torch.Size([84])
fc3.weight torch.Size([10, 84])
fc3.bias torch.Size([10])
```

This will make the clients crash right after connecting to the server.

A workaround could be to "warmup" the model so it sees at least one batch. This will make `num_batches_tracked` be  `tensor(1)`, not causing the error above:

```
# Adding this right after model instantiation 
rnd = torch.randn(3, 3, 32,32) # random tensor of shape = 3x CIFAR10 images
with torch.no_grad():
    net(rnd)
```

**Comment by yasar-rehman** [comment]
I had the same issue. 

I think the problem is in the following line of code in the set_parameter() function:

state_dict = OrderedDict({k: torch.Tensor(v) for k, v in params_dict})

After converting the torch.Tensor(v) to torch.from_numpy() as below, my code executed without any errors.

state_dict = OrderedDict({k: torch.from_numpy(v) for k, v in params_dict})
